src/decoder_allocators.hpp

```diff
@@ -0,0 +1,149 @@
+/*
+    Copyright (c) 2007-2015 Contributors as noted in the AUTHORS file
+
+    This file is part of libzmq, the ZeroMQ core engine in C++.
+
+    libzmq is free software; you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    As a special exception, the Contributors give you permission to link
+    this library with independent modules to produce an executable,
+    regardless of the license terms of these independent modules, and to
+    copy and distribute the resulting executable under terms of your choice,
+    provided that you also meet, for each linked independent module, the
+    terms and conditions of the license of that module. An independent
+    module is a module which is not derived from or based on this library.
+    If you modify this library, you must extend this exception to your
+    version of the library.
+
+    libzmq is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+    License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef ZEROMQ_DECODER_ALLOCATORS_HPP
+#define ZEROMQ_DECODER_ALLOCATORS_HPP
+
+#include <stdlib.h>
+
+#include "err.hpp"
+#include "atomic_counter.hpp"
+
+namespace zmq
+{
+    // Static buffer policy.
+    class c_single_allocator
+    {
+    public:
+        c_single_allocator(size_t bufsize_):
```

Comment:
Ah, we are all maintainers. If the code is readable and works as expected,
that is already a win.

On Sun, Jul 5, 2015 at 11:59 PM, Jens Auer <notifications@github.com> wrote:

> In src/decoder_allocators.hpp
> <https://github.com/zeromq/libzmq/pull/1466#discussion_r33899445>:
>
> > +
> > +#ifndef ZEROMQ_DECODER_ALLOCATORS_HPP
> > +#define ZEROMQ_DECODER_ALLOCATORS_HPP
> > +
> > +#include <stdlib.h>
> > +
> > +#include "err.hpp"
> > +#include "atomic_counter.hpp"
> > +
> > +namespace zmq
> > +{
> > +    // Static buffer policy.
> > +    class c_single_allocator
> > +    {
> > +    public:
> > +        c_single_allocator(size_t bufsize_):
>
> To be honest, as a pure C++-programmer (I do not write C code), I found
> the code-base in some places rather unusual, e.g. using malloc/free and
> C-style casts abundantly. Sometimes, it looks C++ design which tries to be
> similar to C-code. Although it is very readable and I like the design.
> Cheers, Jens
> … <#14e603ad251d865d_>
> On Sunday 05 July 2015 14:47:53 Thomas K�ppe wrote: But don't take
> anything I say too seriously, I'm just an innocent bystander who knows some
> C++ and mostly takes his cues from the existing code base here. Defer to
> the maintainers for all questions regarding style.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeromq/libzmq/pull/1466/files#r33899445>.
>